lib/customer-deposit-wallet/test/scenario/Test/Scenario/Wallet/Deposit/Exchanges.lhs.md

```diff
@@ -0,0 +1,218 @@
+# Use case: Centralized Exchange
+
+This document describes how a centralized exchange (CEX) can use the Deposit Wallet to
+
+1. Assign an address to a customer ID.
+2. Track deposits at this address.
+3. Track deposits at all addresses.
+4. Create payments to a different wallet.
+
+# Scenarios, Haskell
+
+In this section, we describe the scenarios using Haskell.
+
+```haskell
+module Test.Scenario.Wallet.Deposit.Exchanges
+    ( scenarioRestore
+    , scenarioStart
+    , scenarioCreateAddressList
+    , scenarioTrackDepositOne
+    , scenarioTrackDepositAll
+    , scenarioCreatePayment
+    ) where
+
+import Prelude
+
+import Cardano.Crypto.Wallet
+    ( XPrv
+    , XPub
+    )
+import Cardano.Wallet.Deposit.IO
+    ( WalletEnv
+    , WalletInstance
+    )
+import Cardano.Wallet.Deposit.Pure
+    ( Customer
+    , TxSummary (..)
+    , ValueTransfer (..)
+    )
+import Cardano.Wallet.Deposit.Read
+    ( Address
+    , Value
+    )
+import Test.Scenario.Blockchain
+    ( ScenarioEnv
+    , ada
+    , assert
+    , payFromFaucet
+    , signTx
+    , submitTx
+    )
+
+import qualified Cardano.Wallet.Deposit.IO as Wallet
+import qualified Data.Map as Map
+```
+
+We use a function `depositFundsAt` to model the act where a customer deposits funds at a given address.
+
+```haskell
+depositFundsAt :: ScenarioEnv -> Address -> Value -> IO ()
+depositFundsAt env address value = payFromFaucet env [(address, value)]
+```
+
+## 0. Start a Wallet
+
+A `WalletInstance` denotes a mutable wallet that is actively synchronizing to the blockchain, continuously writes its state to a database file, and responds to queries.
+
+In order to create a fresh wallet, or in order to restore a wallet from its public key all the way from genesis, use the function `withWalletInit`. In addition to the public key, this function expects a number which equals the numerically largest customer ID previously handled with this wallet.
```

Comment:
```suggestion
In order to create a fresh wallet, or in order to restore a wallet from its public key all the way from genesis, use the function `withWalletInit`. In addition to the public key, this function expects a number which equals numerically to the largest customer ID previously handled with this wallet.
```

Reply:
* "to equal the numerically largest customer ID" — verb
* "is equal to the numerically largest customer ID" — adjective

Both forms of "equal" are acceptable in English. I often use the verb as it plays better with the active voice.